### ISSUE TITLE
Anchor doesn't work properly after "Set canvas size" action

### ISSUE THREAD

**Issue by dop2000** [Scirra/Construct-bugs#6871]
## Attach a .c3p

[Anchor_bug.zip](https://github.com/Scirra/Construct-bugs/files/11146434/Anchor_bug.zip)

## Steps to reproduce

1. Run the project, try switching to the second layout an back - notice that the anchored objects stay in place.
2. Click "Change canvas" button - anchored objects still work correctly.
3. Switch to another layout and back - red anchored objects now shifted.

## Expected result

Anchor should work correctly when the viewport is changed.

## More details

Please don't say this is by design...

## System details

<details><summary>View details</summary>

Platform information
Product: Construct 3 r335 (beta)
Browser: Chrome 111.0.5563.147
Browser engine: Chromium
Context: browser
Operating system: Windows 10
Device type: desktop
Device pixel ratio: 1.5
Logical CPU cores: 16
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/111.0.0.0 Safari/537.36
Language setting: en-US

Local storage
Storage quota (approx): 272 gb
Storage usage (approx): 1.3 gb (0.5%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc. (AMD)
Renderer: ANGLE (AMD, AMD Radeon(TM) Graphics Direct3D11 vs_5_0 ps_5_0, D3D11)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_color_buffer_half_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_compression_bptc
EXT_texture_compression_rgtc
EXT_texture_filter_anisotropic
EXT_texture_norm16
KHR_parallel_shader_compile
OES_draw_buffers_indexed
OES_texture_float_linear
OVR_multiview2
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
WEBGL_multi_draw
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

WebM Opus (audio/webm; codecs=opus)
Ogg Opus (audio/ogg; codecs=opus)
WebM Vorbis (audio/webm; codecs=vorbis)
Ogg Vorbis (audio/ogg; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM AV1 (video/webm; codecs=av01.0.00M.08)
MP4 AV1 (video/mp4; codecs=av01.0.00M.08)
WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.265 (video/mp4; codecs=hev1.1.2.L93.B0)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
</details>

**Comment by AshleyScirra** [comment]
I checked the code history, and ever since C2, the Anchor behavior has worked based on the original viewport size - i.e. what it says in Project Properties - rather than the current viewport size. I don't know why this is, but given that changing it seems to risk breaking backwards compatibility, we made it over a decade before anyone reported it, the fact the Anchor behavior is just a helper for the system viewport expressions which let you position things relative to the viewport with more control anyway, and the fact we have been inundated with so many issue reports lately that we are struggling to keep up, I don't think we will take action on this. I'd advise to just use the system viewport expressions instead of the Anchor behavior for more advanced cases like this.
